apis/mutations/v1alpha1/assign_types.go

```diff
@@ -54,6 +54,7 @@ type Parameters struct {
 	// is merged, we can use an actual object for `Assign`
 
 	// Assign.value holds the value to be assigned
+	// +kubebuilder:validation:XPreserveUnknownFields
```

Comment:
Why do we need to add preserve unknown fields here?

Reply:
There was an issue where the upgrader is omitting unknown fields such as `spec.parameters.assign.externalData` when upgrading from v1alpha1 since the field is only available for `v1beta1`.

Reply:
I was wondering if there is a better way to handle this.

Reply:
Has the CRD been updated prior to upgrader starting to run?

Are both `v1alpha1` and `v1beta1` being updated to support external data? (They should both support it according to KRM backwards compatibility reqs: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md )

Reply:
@chewong  NOTE: If users are using helm to install/upgrade, then the upgrade crds hook should take care updating existing crds https://github.com/open-policy-agent/gatekeeper/blob/master/manifest_staging/charts/gatekeeper/templates/upgrade-crds-hook.yaml#L91-L93